src/main/java/edu/harvard/iq/dataverse/api/Pids.java

```diff
@@ -104,6 +104,12 @@ public Response reservePid(@PathParam("id") String idSupplied) {
     public Response deletePid(@PathParam("id") String idSupplied) {
         try {
             Dataset dataset = findDatasetOrDie(idSupplied);
+            //Restrict to never-published datasets (that should have draft/nonpublic pids). The underlying code will invalidate
+            //pids that have been made public by a pid-specific method, but it's not clear that invalidating such a pid via an api that doesn't
+            //destroy the dataset is a good idea.
+            if(dataset.isReleased()) {
+            	return badRequest("Not allowed for Datasets that have been published.");
```

Comment:
This "isReleased" check is making me question if we should remove this "delete PID" API endpoint altogether (as well as the DeletePidCommand and docs, of course).

I added it because I was looking at DataCite APIs from a CRUD perspective. How can I create a PID, read it, update it, delete it. I was learning how to do these things with curl and I'm aware that sometimes sysadmins have a need to operate on the DataCite API directly to fix certain things. My thought was that we could make it easier by giving them APIs in Dataverse to operate on DataCite, rather than re-documenting the DataCite API or simply sending them to those docs.

The other thing I was thinking when adding this Delete PID API is that I wanted a way to blank out `GlobalIdCreateTime` and `IdentifierRegistered` so I could test the "register PID" API (which I still think is a useful, necessary API). This is much more of a dev concern than prod, however.

Reply:
I would agree that having an API just for dev or test should either be removed or moved to a different area of the API (I think we have admin/test? we could also add admin/dev?)

Reply:
Moving this, or all of these calls to admin, and then potentially removing the !isReleased() restriction could make sense - if anyone has been destroying datasets with earlier versions of Dataverse, a way to sync up with current practice might be helpful.

Reply:
is there a best practice about only having isSuperUser() restrictions for admin api calls? (I seem to remember that argument from something I did, but not sure if it was a general practice.)

Reply:
they don't have isSuperUser() in the code, because they don't require a token - things like indexing, etc. They are protected by security policy in that they can only be run from localhost. (they also don't use commands, since commands require user / permission checks).

Reply:
There are actually a bunch in Admin.java (some from me) but maybe they're not needed. My question was more in the other direction and taking a look, I see there are lots of non-admin api calls that have a superuser restriction. (If there weren't it might add to the argument that the Pids api should be admin.)

Reply:
I guess I should have said - api calls in admin *should be* for things that don't require a user, and are protected by policy. There are definitely a lot of non admin calls that do have a superuser restriction. Things like Destroy. We want superusers, who don't have localhost access, to be able to call these.

Reply:
As I mentioned to @qqmyers today, I found the "delete PID" API useful today in development. Also, it's used in the testCannotPublishUntilReserved test in PidsIT.